### ISSUE TITLE
List files/folders API endpoint

### ISSUE THREAD

**Issue by olegmoz** [artipie/artipie#545]
We need to implement API for reading files metadata. There is such API endpoint in Artifactory API: https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-FileList

We need it to list folders, and for the start let's assume that listing should not be recursive (e.g. one level of depth)

**Comment by g4s8** [comment]
@olegmoz I'll copy relevant API spec here from Artifactory docs:

**Description**: Get a flat (the default) or deep listing of the files and folders (not included by default) within a folder.
For deep listing you can specify an optional depth to limit the results.
Optionally include a map of metadata timestamp values as part of the result (only properties are displayed in since 3.0.0).
folder inclusion since 2.3.2; checksum inclusion since: 2.3.3; include folder root path since: 2.5.2. Supported by all types of repositories.
**Usage**: `GET /api/storage/{repoKey}/{folder-path}?list[&deep=0/1][&depth=n][&listFolders=0/1][&mdTimestamps=0/1][&includeRootPath=0/1]`
 - `deep` - if `1`, then respect `depth` query param, `0` by default
 - `depth` - `hierarchy depth, use only if `deep` query param is `1`
 - `listFolders` - include folders if `1`, `0` by default
 - `mdTimestamps` - include timestampts
 - `includeRootPath` - include root path item `/`

**Produces**: `application/json (application/vnd.org.jfrog.artifactory.storage.FileList+json)`

**Sample Output**:
`GET /api/storage/libs-release-local/org/acme?list&deep=1&listFolders=1&mdTimestamps=1`
```json
{
"uri": "http://localhost:8081/artifactory/api/storage/libs-release-local/org/acme",
"created": "ISO8601",
"files" : [
  {
  "uri": "/archived",
  "size": "-1",
  "lastModified": "ISO8601",
  "folder": "true"
  },
  {
  "uri": "/doc.txt",
  "size": "253207",
  "lastModified": "ISO8601",
  "folder": "false",
  "sha1": "sha1Checksum",
  "mdTimestamps": { "properties" : "lastModified (ISO8601)" }
  },
  {
  "uri": "/archived/doc1.txt",
  "size": "253100",
  "lastModified": "ISO8601",
  "folder": "false",
  "sha1": "sha1Checksum",
  "mdTimestamps": { "properties" : "lastModified (ISO8601)" }
  }
]
}
```

**Comment by 0crat** [comment]
You @olegmoz[/z](https://www.0crat.com/u/olegmoz) need to have one of these roles in order to do "Request order start": `ARC`, `PO`; I'm sorry to say this, but at the moment you've got no roles in this project

**Comment by olegmoz** [comment]
For now we need to support only requests with `listFolders=1` and `depth=1`

**Comment by 0pdd** [comment]
@olegmoz the puzzle [#597](https://github.com/artipie/artipie/issues/597) is still not solved.

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#545` is not assigned, can't get performer

**Comment by 0pdd** [comment]
@olegmoz the puzzle [#786](https://github.com/artipie/artipie/issues/786) is still not solved; solved: [#597](https://github.com/artipie/artipie/issues/597).

**Comment by 0pdd** [comment]
@olegmoz all 2 puzzles are solved here: [#597](https://github.com/artipie/artipie/issues/597), [#786](https://github.com/artipie/artipie/issues/786).
